lib/compiler/doc/src/compile.xml

```diff
@@ -772,29 +772,44 @@ module.beam: module.erl \
           </item>
         </taglist>
 
-	<p>Another class of warnings is generated by the compiler
-	  during optimization and code generation. They warn about
-	  patterns that will never match (such as <c>a=b</c>), guards
-	  that always evaluate to false, and expressions that
-	  always fail (such as <c>atom+42</c>).</p>
-	  <p>Those warnings cannot be disabled (except by
-	  disabling all warnings).</p>
+	<p>Another kind of warnings is <em>opportunistic
+	warnings</em>. They are generated when the compiler
+	<strong>happens</strong> to notice potential issues during
```

Comment:
Should this stress "potential" and not "happens"? Or is it not a given that in some cases e.g. unused returned values would be missed by the compiler?

Reply:
I wanted to emphasize the other kind of problem with opportunistic warnings, namely that not all potential problems are reported. I think it is better to remove the emphasis on "happens".